processor/k8sprocessor/kube/client.go

```diff
@@ -386,3 +468,42 @@ func selectorsFromFilters(filters Filters) (labels.Selector, fields.Selector, er
 	}
 	return labelSelector, fields.AndSelectors(selectors...), nil
 }
+
+func (c *WatchClient) addOrUpdateNamespace(namespace *api_v1.Namespace) {
+	newNamespace := &Namespace{
+		Name:         namespace.Name,
+		NamespaceUID: string(namespace.UID),
+		StartTime:    namespace.GetCreationTimestamp(),
+	}
+	newNamespace.Attributes = c.extractNamespaceAttributes(namespace)
+
+	c.m.Lock()
+	if namespace.Name != "" {
+		c.Namespaces[namespace.Name] = newNamespace
+	}
+	c.m.Unlock()
+}
+
+func (c *WatchClient) forgetNamespace(namespace *api_v1.Namespace) {
+	c.m.Lock()
+	if ns, ok := c.Namespaces[namespace.Name]; ok {
+		delete(c.Namespaces, ns.Name)
```

Comment:
Can you please add a comment here describing why this does not need the grace period support like pods have (with deleteQueue)

Reply:
Why do we need the grace period implementation for pods? I didnt see any comments explicitly mentioning why we need it?
If I know that I can add a comment for the namespace stuff too :)

Reply:
So I think this wouldnt happen with namespaces. Its because when a namespace is deleted all the pods(and other k8s objects in that namespace) in that namespace are deleted before it. So we wont have any spans that might need namespace annotations and labels. Is this cool? I can add it as a comment if you recommend it.

Reply:
I added the following
`			// When a namespace is deleted all the pods(and other k8s objects in that namespace) in that namespace are deleted before it.
			// So we wont have any spans that might need namespace annotations and labels.
			// Thats why we dont need an implementation for deleteQueue and gracePeriod for namespaces.`

Reply:
It might still happen if the spans are delayed but very unlikely. comment looks fine.